sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -287,11 +288,14 @@ abstract class BaseHashJoinIterator(
   }
 
   override def createGatherer(
-      cb: ColumnarBatch,
+      streamBatch: LazySpillableColumnarBatch,
       numJoinRows: Option[Long]): Option[JoinGatherer] = {
     try {
-      withResource(GpuProjectExec.project(built.getBatch, boundBuiltKeys)) { builtKeys =>
-        joinGatherer(builtKeys, built, cb)
+      streamBatch.allowSpilling()
+      withRetryNoSplit {
+        withResource(GpuProjectExec.project(built.getBatch, boundBuiltKeys)) { builtKeys =>
+          joinGatherer(builtKeys, built, streamBatch)
```

Comment:
This call could close one of the batches.  If it closes the built batch we are ok, but if it closes the streamBatch, that will break on retries.